File: Daily Test/C++/Longest SubArray - Integer Train 4.2.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<string> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];

    int bestLen = 0, bestSum = 0;

    for (int i = 0; i < n - 1; ) {
        int start = i;
        int sum = stoi(arr[i]);

        while (i < n - 1 && arr[i].back() == arr[i + 1].front()) {
            sum += stoi(arr[i + 1]);
            i++;
        }

        int len = i - start + 1;
        if (len >= 2) {
            if (len > bestLen || (len == bestLen && sum > bestSum)) {
                bestLen = len;
                bestSum = sum;
            }
        }
        i++;
    }

    if (bestLen == 0) {
        int mx = 0;
        for (auto &s : arr) mx = max(mx, stoi(s));
        cout << mx;
    } else {
        cout << bestSum;
    }
}

File: Daily Test/Java/Longest SubArray - Integer Train 4.2.2026.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] arr = new String[n];
        for (int i = 0; i < n; i++) arr[i] = sc.next();

        int bestLen = 0, bestSum = 0;

        for (int i = 0; i < n - 1; ) {
            int start = i;
            int sum = Integer.parseInt(arr[i]);

            while (i < n - 1 && 
                   arr[i].charAt(arr[i].length() - 1) == arr[i + 1].charAt(0)) {
                sum += Integer.parseInt(arr[i + 1]);
                i++;
            }

            int len = i - start + 1;
            if (len >= 2) {
                if (len > bestLen || (len == bestLen && sum > bestSum)) {
                    bestLen = len;
                    bestSum = sum;
                }
            }
            i++;
        }

        if (bestLen == 0) {
            int mx = Integer.MIN_VALUE;
            for (String s : arr)
                mx = Math.max(mx, Integer.parseInt(s));
            System.out.println(mx);
        } else {
            System.out.println(bestSum);
        }
    }
}